init.lua

```diff
@@ -44,60 +80,435 @@ P.S. You can delete this when you're done too. It's your config now :)
 vim.g.mapleader = ' '
 vim.g.maplocalleader = ' '
 
+-- [[ Setting options ]]
+-- See `:help vim.opt`
+-- NOTE: You can change these options as you wish!
+--  For more options, you can see `:help option-list`
+
+-- Set highlight on search
+vim.opt.hlsearch = false
```

Comment:
Can you elaborate on why this is a good default? Doesn't this hide useful information when searching and hitting `n` to go to the next search.

Reply:
I like this combination as a better default:

```lua
-- Set highlight on search
vim.opt.hlsearch = true

-- [[ Highlight on yank ]]
-- See `:help vim.highlight.on_yank()`
vim.api.nvim_create_autocmd('TextYankPost', {
  group = vim.api.nvim_create_augroup('YankHighlight', { clear = true }),
  callback = function()
    vim.highlight.on_yank()
  end,
})

vim.keymap.set('n', '<CR>', ':nohlsearch<CR>', { silent = true })

```